microcosm_flask/swagger/definitions.py

```diff
@@ -120,21 +121,53 @@ def iter_definitions(definitions, operations):
         yield get_response_schema(func)
 
 
-def build_path(operation, ns):
+def build_path_for_integer_param(ns, operation, arguments):
     """
-    Build a path URI for an operation.
+    Build an endpoint path when the parameters are integer-valued
+
+    When building paths for swagger, parameter names are substituted for
+    path parameters. For example, the output will have '/api/v1/person/{person_id}'.
+    We still use werkzeug to build those paths with placeholders for params.
+
+    That works with UUID-valued ids but not integer-valued ones, due do a difference
+    between the UUIDConverter and NumberConverter's `to_url` methods
+    (see https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L1315
+    and https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L1234)
+
+    Here we instead use placeholder integers to use werkzeug functions, and replace
+    them afterwards.
 
     """
+    sorted_args = list(arguments)
+    args_substitution = {
+        PLACEHOLDER_INTEGER_ID + index: argument
+        for index, argument in enumerate(sorted_args)
+    }
+    uri_templates = {
+        argument: f"{placeholder}"
+        for placeholder, argument in args_substitution.items()
+    }
+    path = unquote(ns.url_for(operation, _external=False, **uri_templates))
+    for placeholder_integer, argument in args_substitution.items():
+        path = path.replace(str(placeholder_integer), f"{{{argument}}}")
+
+    return path
+
+
+def build_path(operation, ns):
     try:
         return ns.url_for(operation, _external=False)
     except BuildError as error:
-        # we are missing some URI path parameters
-        uri_templates = {
-            argument: "{{{}}}".format(argument)
-            for argument in error.suggested.arguments
-        }
-        # flask will sometimes try to quote '{' and '}' characters
-        return unquote(ns.url_for(operation, _external=False, **uri_templates))
+        if ns.identifier_type == "int":
+            return build_path_for_integer_param(ns, operation, error.suggested.arguments)
```

Comment:
Can this be achieved before the error is thrown or is the only place to get the arguments? Also does the `error.suggested.arguments` contain all the arguments or just the ones that had issues? If the latter are non-problematic arguments correctly handled?

Reply:
`error.suggested.arguments` only contains the params that had issues. In this context however, all param will error because we're not passing in values for any of them.

The idea is that we're asking werkzeug to build a URL like `/api/v1/person/:person_id` but we don't have a `person_id`. So we let werkzeug tell us what's missing, and put the param name in there (e.g. `{person_id}`); and we end up with the right format for OpenAPI.

Reply:
Does `microcosm_flask` support  building urls such as: `/api/v1/users/<uuid>/company/<int>/queue/<int>/`? As it looks like these will fail to build correctly as code relies on no nesting of various types in a single url.

Reply:
yeah i don't think we do doulbe uuid urls.

Reply:
Right, we mostly just go one layer deep (e.g. `/api/v1/company/<id>/event`), when we do use that convention.